### ISSUE TITLE
Mounting REAL cdrom in Guest OS !?

### ISSUE THREAD

**Issue by angelinocapobianco** [joncampbell123/dosbox-x#3409]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

In my config i mount my local real cdrom drive with "MOUNT D E:\ -t cdrom".
It Work in Dosbox-X but when i start the Windows 98, there is no way to see the cdrom.

If i mount ISO file it work, but not with the real CDROM.

There is a way?

**Comment by Wengier** [comment]
Indeed, only image files are visible once you boot into guest systems, so try mount ISO files in such case.

**Comment by JollyWizard** [comment]
I was looking at this today, because I didn't expect it to be a limitation.

I have read multiple sources saying that it doesn't work, but others seeming to suggest using `-ide 2m`.

So I tried using this command: `mount D "D:\" -ide 2m`.

The files are visible from dos, but not Win98SE.

I moved my mount command to be before `imgmount C ...` and `boot c`. 

I am now seeing my mounted real DVD-Drive inside Win98, but it shows up as a FAT folder. The autorun installer triggers for a Tonka Fire and Rescue CD I have, but it says that "No CD drive is detected."

**Comment by rderooy** [comment]
What you are trying to do will not work.

What you have managed to do is this:
https://dosbox-x.com/wiki/Guide%3AManaging-image-files-in-DOSBox%E2%80%90X#_creating_harddisk_images_from_folder_mounts

**Comment by rderooy** [comment]
Sorry, meant to link to this:
https://dosbox-x.com/wiki/Guide%3AInstalling-Windows-98#_host_folder_mounts

**Comment by Torinde** [comment]
Yeach, so this issue should be tagged with "feature request"...

You may try also the "-usecd" options. I think that will make your physical host CD-ROM drive to appear inside DOSbox as CD-ROM. Somebody wrote that probably Win9x can run from the DOSbox-X shell, without booting MS-DOS, but I haven't tested that...

```
Mounts directories or drives in the host system as DOSBox-X drives.
Usage: MOUNT drive local_directory [option]
 drive            Drive letter where the directory or drive will be mounted.
 local_directory  Local directory or drive in the host system to be mounted.
 [option]         Option(s) for mounting. The following options are accepted:
 -t               Specify the drive type the mounted drive to behave as.
                  Supported drive type: dir, floppy, cdrom, overlay
 (Note that 'overlay' redirects writes for mounted drive to another directory)
 -label [name]    Set the volume label name of the drive (all upper case).
 -nl              Use next available drive letter if the drive is mounted.
 -ro              Mount the drive in read-only mode.
 -pr              Specify the path is relative to the config file location.
 -cd              Generate a list of local CD drive's "drive #" values.
 -usecd [drive #] For direct hardware emulation such as audio playback.
 -ioctl           Use lowest level hardware access (following -usecd option).
 -aspi            Use the installed ASPI layer (following -usecd option).
 -freesize [size] Specify the free disk space of drive in MB (KB for floppies).
 -nocachedir      Enable real-time update and do not cache the drive.
 -z drive         Move virtual drive Z: to a different letter.
 -o               Report the drive as: local, remote.
 -q               Quiet mode (no message output).
 -u               Unmount the drive.
 -examples        Show some usage examples.
Type MOUNT with no parameters to display a list of mounted drives.
```

> I can't insert a disk and have it updated

Try Drives \ D \ Rescan drive
![image](https://github.com/joncampbell123/dosbox-x/assets/97228894/b81593c2-92ac-4b58-ac5b-8cede935abf5)

**Comment by rderooy** [comment]
You cannot run Win98 from the dosbox-x shell. You can only get through the first phase of the installer, not actually run Win98.

For `-usecd`, if I recall correctly, that basically does the equivalent of a `mount`, so it will not survive booting to a guest OS, and it will also not include any redbook audio tracks the CD may have.

**Comment by JollyWizard** [comment]
> What you are trying to do will not work.
> 
> What you have managed to do is this: https://dosbox-x.com/wiki/Guide%3AManaging-image-files-in-DOSBox%E2%80%90X#_creating_harddisk_images_from_folder_mounts

> Sorry, meant to link to this: https://dosbox-x.com/wiki/Guide%3AInstalling-Windows-98#_host_folder_mounts

> You cannot run Win98 from the dosbox-x shell. You can only get through the first phase of the installer, not actually run Win98.
> 
> For `-usecd`, if I recall correctly, that basically does the equivalent of a `mount`, so it will not survive booting to a guest OS, and it will also not include any redbook audio tracks the CD may have.

I looked at this documentation and didn't feel they documented the use case when you are pointing to a CD drive.  

As you are saying, it is performing a host folder mount, and will not work with redbook, but you can put a CD before boot and access the data. 

My main purpose in commenting is to bring awareness to the fact that this use case is partially supported, whereas the documentation and threads I've seen were not clear.

> Try Drives \ D \ Rescan drive

Not supported. See Pic.
![image](https://github.com/joncampbell123/dosbox-x/assets/4997950/6f84049b-2e31-4cbc-9ddc-810dbe5d2608)

> Yeach, so this issue should be tagged with "feature request"...

I would say yes, but also a documentation update. There are clear use cases here that are not documented and only have partial support.

User Stories:
1. User has a CD with data (or installers) that they want to access, which DO NOT DEPEND on CD drive detection or Red Book audio.  The user must mount the CD drive as a folder mount before booting the Guest OS.
2. User has a CD with data (or installers) that they want to access, which DO DEPEND on CD Drive detection or Red Book Audio. This use case is not currently supported.

Partial/Full CD ROM support in the guest os could take several forms.

Potential Feature Requests:
* Allow rescanning folder mounts after guest os boot.
* Allow mounted read only drives to appear as a data only CDROM to guest OS.
* Allow actual CDROM data to be seen as a cdrom instead of a fat drive.
* Allow actual CDROMS to be automatically rescanned after a disc change.
* Allow actual CDROMS to provide access to Redbook audio.
* Clone mounted cd drives data to a mounted image instead of converting to FAT on Guest OS boot. 
 
The clone to mounted image option is not ideal, but is the path taken by the Mister community since they can't support cartridge readers.

A clone to image solution could piggyback off the existing systems, using only gui / utility side features, which would be convenient to implement, but the time intensity could catch users by surprise if not fully thought out.  

I have 280 mb data disc spinning down from apparent caching right around the time the OS is ready to open My Computer.  An imgburn `bin` rip for my `Frogger 2` disc with audio tracks took something like 5 minutes or more.  Preferably, you would have some sort of caching option that can match the current drive disc to a ripped image. At that point you are looking at managing an image collection as an improvement, and I think that would play nicer with the disc swap feature as is and be a better route to take than the Dosbox-Pure convention of prepackaging everything in advance.  The main thing I like about Dosbox-X is that it is more configurable from the gui and after boot.
